Lib/fontmake/font_project.py

```diff
@@ -420,6 +450,33 @@ def _output_path(self, ufo, ext, is_instance=False, interpolatable=False,
             os.makedirs(out_dir)
         return os.path.join(out_dir, '%s.%s' % (self._font_name(ufo), ext))
 
+    def _designspace_locations(self, designspace_path):
+        """Map font filenames to their locations in a designspace."""
+
+        maps = []
+        for location_list in varLib.designspace.load(designspace_path):
+            location_map = {}
+            for loc in location_list:
+                abspath = os.path.normpath(os.path.join(
+                    os.path.dirname(designspace_path), loc['filename']))
+                location_map[abspath] = loc['location']
+            maps.append(location_map)
+        return maps
+
+    def _closest_location(self, location_map, target):
+        """Return path of font whose location is closest to target."""
+
+        dist = lambda a, b: math.sqrt(sum((a[k] - b[k]) ** 2 for k in a.keys()))
```

Comment:
So I was confused about what normalization meant. But Doug and I still agreed that we don't need to normalize here as long as the ranges of the axes are the same. We can revisit this if it turns out that's not a safe assumption.